pkg/webhook/sidecar_spec.go

```diff
@@ -74,15 +76,36 @@ func GetSidecarContainerSpec(c *Config) v1.Container {
 			},
 		},
 	}
+
+	if addCacheVolume {
+		toReturn.VolumeMounts = append(toReturn.VolumeMounts, v1.VolumeMount{
+			Name:      CacheVolumeName,
+			MountPath: CacheVolumeMountPath,
+		})
+	}
+	return toReturn
 }
 
-func GetSidecarContainerVolumeSpec() v1.Volume {
-	return v1.Volume{
-		Name: SidecarContainerVolumeName,
-		VolumeSource: v1.VolumeSource{
-			EmptyDir: &v1.EmptyDirVolumeSource{},
+func GetSidecarContainerVolumeSpec(addCacheVolume bool) []v1.Volume {
+	toReturn := []v1.Volume{
+		{
+			Name: SidecarContainerVolumeName,
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: &v1.EmptyDirVolumeSource{},
+			},
 		},
 	}
+	if addCacheVolume {
+		toReturn = append(toReturn, v1.Volume{
+			Name: CacheVolumeName,
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: "/mnt/stateful_partition/kube-ephemeral-ssd",
```

Comment:
Could you add a comment here to mention that we assume the local ssd is enabled on the node, and this change is currently for testing?

Reply:
Resolving as discussed offline. This feature no longer necessitates driver changes.